pandas/core/common.py

```diff
@@ -381,25 +401,17 @@ def _asarray_tuplesafe(values, dtype=None):
         return values.values
 
     if isinstance(values, list) and dtype in [np.object_, object]:
-        return lib.list_to_object_array(values)
+        return _as_1d_array(values)
 
     result = np.asarray(values, dtype=dtype)
 
     if issubclass(result.dtype.type, compat.string_types):
         result = np.asarray(values, dtype=object)
 
     if result.ndim == 2:
-        if isinstance(values, list):
-            return lib.list_to_object_array(values)
```

Comment:
so is seems like list basically replaces ``lib.list_to_object_array`` if that is so, then we should simply do that (it can live in pandas.core.dtypes.cast as well).

```
if hasattr(values, '__array__'):
             values = [tuple(x) for x in values     
result = _as_1d_array(values)
```

Reply:
Yes, ``_as_1d_array`` can entirely replace ``lib.list_to_object_array``. My only doubt is that the latter is used in some cython files... I have no idea if in those cases using a cython function has advantages.

Reply:
Sorry, I had missed the second part of your comment.And I don't understand it (isn't it what I'm doing?)

Reply:
well, this essentially replaces ``list_to_object``, so if you can do it and it is just as fast I don't care if its in cython or not. So go ahead and generally replace it (if it will do the job). Run asv's for this.

See my comments on the code location, these need to be named similarly to other functions, and live in pandas.core.dtypes.cast.